docs/release.rst

```diff
@@ -25,6 +25,8 @@ Fixes
   ``crc32c`` is not installed. This has been changed to match
   the behaviour of other optional dependencies/codecs.
   By :user:`John Kirkham <jakirkham>`, :issue:`637`
+* Fixes issue with ``Delta`` Zarr 3 codec not working with ``astype``.
```

Comment:
This should move up into a new unreleased section (sorry, I forgot to add in a new section when I released 0.14.1)